### PR TITLE
Implementation details of punishment module

### LINKED ISSUE #84: Problem: punishments are not defined precisely
define precisely (somewhat "formally") in relation to Tendermint ABCI what each part does, so that this can be easily checked against the implementation:
what are data inputs from where/when
how are they processed
what are outputs and when they are emitted

### PR DESCRIPTION
**Pull request by devashishdxt**
Fixes #84.

### CODE REVIEW

**Review by yihuang — APPROVED** [review]

File: docs/getting-started/punishments.md [hunk 1]
```diff
@@ -105,3 +105,137 @@ evidence that we get from tendermint and ignore other evidences until `UNBONDING
...
+
+#### `PunishmentConfig`
+
+`PunishmentConfig` holds the values for all the network parameters related to punishment. `PunishmentConfig` can be
+initialized from `genesis.json`.
+
+```rust
+/// Configuration for computing and executing validator punishments
+pub struct PunishmentConfig {
+    /// Number of blocks for which the moving average is calculated for uptime tracking.
```
Comment (tomtau) [inline docs/getting-started/punishments.md]: moving average?

**Review by tomtau — COMMENTED** [review]
looks ok

**Review by lezzokafka — APPROVED** [review]
Looks good to me.
May I ask wha's our plan for the [evidence checking](https://github.com/crypto-com/chain-docs/pull/96/files#diff-8f9dcaf1e4318fb72804d9070866d362R220)?

**Comment by devashishdxt** [comment]
> Looks good to me.
> May I ask wha's our plan for the [evidence checking](https://github.com/crypto-com/chain-docs/pull/96/files#diff-8f9dcaf1e4318fb72804d9070866d362R220)?

I'm not sure we can do anything except for checking evidence's height. Tendermint does all the evidence verification.

File: docs/getting-started/punishments.md (hunk 1 above)
Comment (devashishdxt) [reply docs/getting-started/punishments.md]: Fixed.

**Pull request merged**